pkg/search/proxy/controller.go

```diff
@@ -228,29 +228,36 @@ func (ctl *Controller) reconcile(util.QueueKey) error {
 				klog.Warningf("cluster %s is notReady", cluster.Name)
 				continue
 			}
-
-			if _, exist := resourcesByClusters[cluster.Name]; !exist {
-				resourcesByClusters[cluster.Name] = make(map[schema.GroupVersionResource]*store.MultiNamespace)
-			}
-
-			for resource, multiNS := range matchedResources {
-				gvk, err := ctl.restMapper.KindFor(resource)
-				if err != nil {
-					klog.Errorf("Failed to get gvk: %v", err)
-					continue
-				}
-				if !helper.IsAPIEnabled(cluster.Status.APIEnablements, gvk.GroupVersion().String(), gvk.Kind) {
-					klog.Warningf("Resource %s is not enabled for cluster %s", resource.String(), cluster)
-					continue
-				}
-				resourcesByClusters[cluster.Name][resource] = multiNS
-			}
+			ctl.reconcileResourcesByClusters(resourcesByClusters, cluster, matchedResources)
 		}
 	}
 
 	return ctl.store.UpdateCache(resourcesByClusters, registeredResources)
 }
 
+func (ctl *Controller) reconcileResourcesByClusters(resourcesByClusters map[string]map[schema.GroupVersionResource]*store.MultiNamespace, cluster *clusterv1alpha1.Cluster, matchedResources map[schema.GroupVersionResource]*store.MultiNamespace) {
```

Comment:
```suggestion
func (ctl *Controller) reconcileResourcesByClusters(resourcesByClusters map[string]map[schema.GroupVersionResource]*store.MultiNamespace, cluster *clusterv1alpha1.Cluster, matchedResources map[schema.GroupVersionResource]*store.MultiNamespace) {
```
Does the name `MergeResourcesByCluster` look better?

Reply:
I'm glad to see that you added a new method which makes it easier to have a unit test.
Can we try to add some unit tests on it this time?

Reply:
> ok!by the way,your suggestion no changes

My bad, updated! Thanks for the reminder.

Reply:
> Do you mean adding unit tests for MergeResourcesByCluster? I saw there are already unit test for reconcile

Oh, right! I didn't notice it. But the unit test missed this bug, right? Shall we add a test to it?

Reply:
That's another story. Now focus on fixing the bug, even if the resourceSelector.namespace uses a slice, this issue also needs to be fixed.

Reply:
Thanks in advance. We probably need to explain the reason and the real-world use case. Welcome to propose a new feature or enhancement if you like.

By the way, are you using karmada-search now?

Reply:
Sounds good!